core/src/services/routing.js

```diff
@@ -740,19 +740,46 @@ class RoutingClass {
       url.search = finalUrl;
     }
     // only if close modal [X] is pressed
-    if (historyState && isClosedInternal) {
+    if (history.state && history.state.modalHistoryLength >= 0 && isClosedInternal) {
+      const modalHistoryLength = history.state.modalHistoryLength;
+      const path = history.state.pathBeforeHistory;
+      let isModalHistoryHigerAsHistoryLength = history.state.isModalHistoryHigerAsHistoryLength;
```

Comment:
**history.state.isModalHistoryHigerAsHistoryLength**:

We access the property **isModalHistoryHigerAsHistoryLength** of history state, but seems we never set its value elsewhere or I missed it somehow ? Where does it come from 🤔